src/cps/pc_compat/pc_loader.hpp

```diff
@@ -0,0 +1,23 @@
+// SPDX-License-Identifier: MIT
+// Copyright © 2024 Haowen Liu
+
+#pragma once
+
+#include "cps/pc_compat/pc.parser.hpp"
+#include <unordered_map>
+
+class PcLoader {
+  public:
+    PcLoader();
+
+    std::unordered_map<std::string, std::string> variables;
+    std::unordered_map<std::string, std::string> properties;
```

Comment:
I'm not sure what `variables` and `properties` mean or what they are used for just looking at the declarations here. Some design notes somewhere would be helpful.

Reply:
To clarify, pkg-config has "properties" which have speced meaning and are separated with a `:` (Cflags, Libs, Name, etc); and "variables" which can be queried or used for interpolation but are otherwise arbatrary and use `=` to separate them.

Reply:
Will add some comments. Tho both of them should probably be private and have `parse` return a property map